### PR TITLE
Feature/updating indeed scraper

### PR DESCRIPTION
**Pull request by sammytheindi**
# Pull Request Template

## Description

Have migrated the backend to use the mobile application endpoints and mobile user agents instead of browser. This was motivated by never having any CAPTCHA requests on the mobile application. Have additionally updated the scraping method to work with the new mobile version of Indeed. This fixes issues we were previously having, particularly with TAGS and REMOTENESS. 

The maximum number of results on the mobile version of the application is 20 instead of 50, so I have changed the default scraping to only scrape the results page, and not the individual pages. We can undo this in the pull request if this is not wanted.

Please note these changes have not (yet) been added to the French and German versions of the scraper.

## Context of change

Please add options that are relevant and mark any boxes that apply.

- [x] Software (software that runs on the PC)
- [ ] Library (library that runs on the PC)
- [ ] Tool (tool that assists coding development)
- [ ] Other

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration.

- [x] Test on Job Search with demo/settings_USA.yaml with empty results folder. Made sure it reads the page correctly, and writes jobs to file.
- [x] Test on Job Search with demo/settings_USA.yaml with filled results folder. Made sure the duplicate handling works correctly.
- [x] Ran 20 consecutive tests with demo/settings_USA.yaml to do a smoke test for CAPTCHA presence. No Captcha was present.

## Checklist:

Please mark any boxes that have been completed.

- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by PaulMcInnis — CHANGES_REQUESTED** [review]
Can you open a seperate pr with just formatting changes / can we align more on the formatting?

open to adding a prettier.rc or whatever works - just I’m having trouble seeing the actual diff. 

If you tell me what formatter you used I can also do this.

**Comment by PaulMcInnis** [comment]
👋  I can see that maybe I made a mistake mentioning prettier earlier, looks like this is using `black` to format

**Comment by sammytheindi** [comment]
Spent a few hours trying to get the formatting correct, but I can't understand why github's compare is not showing the correct changes. Formatted base branch is on PR #167. The files that have been changed (I use GitLens' search and compare) are shown below:

Screenshot 2024-09-12 at 1 22 44 PM

**Comment by PaulMcInnis** [comment]
OK @sammytheindi I figured it out, turns out that github PR diffs are crappy and the way to fix this is to do a `git merge master` on this branch so that it uses the correct base branch.

https://github.com/PaulMcInnis/JobFunnel/tree/166-update-indeed-scraper-by-using-mobile feel free to steal commits form here

File: jobfunnel/__init__.py
Comment (PaulMcInnis) [inline jobfunnel/__init__.py:3]: can we up the version to 4.0.0

**Comment by sammytheindi** [comment]
Okay, good to go! If you can pull the most recent changes from `master` into the `indeed_fixes` which is the base branch for this PR, then the files should show (mostly) the correct changes.

**Review by PaulMcInnis — APPROVED** [review]

**Comment by PaulMcInnis** [comment]
I guess next we'll open a PR from indeed fixes into master

**Comment by sammytheindi** [comment]
> I guess next we'll open a PR from indeed fixes into master

Indeed ;) 

I don't have write access to merge this, so I believe you need to merge it in?

**Comment by PaulMcInnis** [comment]
Done, I’ve also invited you to be a collaborator which will let you open and merge prs (with approval)

**Pull request merged**
